### PR TITLE
Fix missing `Field.default` when it's value is `None` in openapi spec

### PR DESCRIPTION
**Pull request by ddorian**
Checklist:

- [x] Run `pytest tests` and no failed.
- [x] Run `ruff check flask_openapi3 tests examples` and no failed.
- [x] Run `mypy flask_openapi3` and no failed.
- [x] Run `mkdocs serve` and no failed.

### CODE REVIEW

File: flask_openapi3/openapi.py [hunk 1]
```diff
@@ -242,6 +242,9 @@ def api_doc(self) -> Dict:
 
         # Convert spec to JSON
         self.spec_json = json.loads(spec.model_dump_json(by_alias=True, exclude_none=True, warnings=False))
+        # fix to include `default=None` in fields
+        components_only = json.loads(spec.model_dump_json(by_alias=True, exclude_unset=True, warnings=False))["components"]
```
[the remark this answers is not in the document]
Comment (ddorian) [reply flask_openapi3/openapi.py]: The other models have many fields with default None that show up in the output (example=None, security=None, deprecated=None etc).

This worked for me on a 5K lines json schema.

[the remark this answers is not in the document]
Comment (luolingchun) [reply flask_openapi3/openapi.py]: I agree to replace `exclude_none` with `exclude_unset`, but I don't think it's perfect just for `components`.
Can you write an example of an error other than the action `components`?

[the remark this answers is not in the document]
Comment (ddorian) [reply flask_openapi3/openapi.py]: Using this:
```python
self.spec_json = spec.model_dump(mode="json", by_alias=True, exclude_unset=True, warnings=False)
```

These are the fields that get removed:
```python
requestBody.required = True
```

These are the extra fields that show up in json. I guess they are being set from functions.
```python
Operation.deprecated = None
Operation.externalDocs = None
Opeation.example = None
Operation.examples = None
Operation.servers = None
Operation.security = None
Parameter.deprecated = None
Parameter.example = None
Parameter.examples = None
```

So we can do my way or fix functions that set the values above to None.

[the remark this answers is not in the document]
Comment (luolingchun) [reply flask_openapi3/openapi.py]: Some models end up in parameters, so just using the unset rule in components is not a good solution.

I committed a commit with all assignments using the unset rule.

Hopefully we can agree.

[the remark this answers is not in the document]
Comment (ddorian) [reply flask_openapi3/openapi.py]: Works great for me.

**Review by ddorian — COMMENTED** [review]
On paths without parameters, parameters shows as `null`, but it's not an issue.

-----

After this, we can add even more defaults to make the code nicer. Like setting:
```python
# instead of 
webhooks: Optional[Dict[str, Union[PathItem, Reference]]] = None
# we do
webhooks: Dict[str, Union[PathItem, Reference]] = {}
# and output should be the same in all cases

```

File: flask_openapi3/openapi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddorian) [reply flask_openapi3/openapi.py]: After my 2 commits my tests are fine and the json is valid openapi spec.

[the remark this answers is not in the document]
Comment (ddorian) [reply flask_openapi3/openapi.py]: @luolingchun can you merge my 2 PRs?

**Pull request merged**